src/wp-includes/compat.php

```diff
@@ -40,6 +40,40 @@ function _wp_can_use_pcre_u( $set = null ) {
 	return $utf8_pcre;
 }
 
+/**
+ * Indicates if a given slug for a character set represents the UTF-8 text encoding.
```

Comment:
Some other functions in this file mention that they're for internal use, like this:

> Internal compat function to mimic mb_substr().

Should we mention that this is an internal function only intended to prevent load-order issues and that `is_utf8_charset` should be used in most cases?

Reply:
I considered this, but also I don't see a reason why this shouldn't be made available 🤷‍♂️

Reply:
`is_utf8_charset` is the API we expect folks to use, not this new function. To me, that's a good reason this should include `@ignore` and be noted as an internal compact function

I also think we need a note for why this is here since it's not a compat function like everything else